cli/azd/pkg/tools/dotnet/dotnet.go

```diff
@@ -166,10 +179,11 @@ func (cli *dotNetCli) PublishAppHostManifest(
 	return nil
 }
 
-// PublishContainer runs a `dotnet publish“ with `PublishProfile=DefaultContainer` to build and publish the container.
+// PublishContainer runs a `dotnet publish“ with `/t:PublishContainer`to build and publish the container.
+// It also gets port number by using `--getProperty:GeneratedContainerConfiguration`
 func (cli *dotNetCli) PublishContainer(
 	ctx context.Context, project, configuration, imageName, server, username, password string,
-) error {
+) (int, error) {
```

Comment:
should we use `string` instead of `int` ?  
I feel like we are not doing anything where the `int` representation is relevant (like doing some math).  We are only reading the value from the output and writing it to another yaml file, right?  So, I think we don't really need the Integer conversion... (unless we want to do validation about the port dotnet is giving us is really a number)

Reply:
My preference would be to keep this as an it.  While we are just writing it to a yaml file and it is a string while in the file, the value represents a number (and the model type itself for the ContainerApp would type this parameter as `int`, so I feel like converting it to an int as soon as possible and keeping it like that is better).